### PR TITLE
feat(node): Add support for symbol-based `CustomPromisify`

### PR DESCRIPTION
**Pull request by ExE-Boss**
Unlike&nbsp;the&nbsp;compiler‑only `__promisify__`&nbsp;property, the&nbsp;`[util.promisify.custom]` symbol&nbsp;property actually&nbsp;exists&nbsp;at&nbsp;runtime.

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/nodejs/node/blob/e001710219906b45bb5ac94d7e9cb4b56a83c134/lib/internal/util.js#L313-L321>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by sandersn** [comment]
@galkin @SimonSchick @Flarna @rbuckton you are frequent node contributors. Can you take a look at this change? I don't know enough about promises to know whether it's good or not.

**Review by SimonSchick — APPROVED** [review]
Fine overall, not quite on board on the naming used here though.

File: types/node/util.d.ts [hunk 1]
```diff
@@ -89,6 +85,16 @@ declare module "util" {
         fn: (arg1: T1, arg2: T2, arg3: T3, arg4: T4, arg5: T5, arg6: T6) => Promise<TResult>
     ): (arg1: T1, arg2: T2, arg3: T3, arg4: T4, arg5: T5, arg6: T6, callback: (err: NodeJS.ErrnoException | null, result: TResult) => void) => void;
 
+    interface CustomPromisifyLegacy<TCustom extends Function> extends Function {
```
[the remark this answers is not in the document]
Comment (ExE-Boss) [reply types/node/util.d.ts]: Well, the&nbsp;new&nbsp;way does&nbsp;work, just&nbsp;currently not&nbsp;with&nbsp;`namespace`:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/96418fa420c42f2d9b1afd8f9b1dfac213cb287c/types/node/node-tests.ts#L133-L150

[the remark this answers is not in the document]
Comment (SimonSchick) [reply types/node/util.d.ts]: I'm still a little bit concerned about usability and how we tell developers how to use this feature.

The _only_ way I can currently create a promisifiable function without casting with your changes is something like this:

```ts
export const myFn = Object.assign(function myFn() {
    return 1;
}, {
    async [util.promisify.custom](): Promise<number> {
        return 1;
    }
});

const promisified: () => Promise<number> = util.promisify(myFn);
console.log(promisified);
promisified().then(console.log);
```

Which yields d.ts:
```ts
import { promisify } from 'util';
export declare const myFn: (() => 1) & {
    [promisify.custom](): Promise<number>;
};
```

How do you envision developers will use `promisify.custom` in the wild?

[the remark this answers is not in the document]
Comment (ExE-Boss) [reply types/node/util.d.ts]: If&nbsp;you&nbsp;do:
```js
export function myFn() {
	return 1;
}
myFn[util.promisify.custom] = async () => {
	return 1;
};
```

You&nbsp;should&nbsp;get:
```ts
import { promisify } from 'util';
export var myFn: {
	(): 1;
	[promisify.custom](): Promise<1>;
};
```

**Comment by ExE-Boss** [comment]
Looks&nbsp;like something&nbsp;broke in&nbsp;`types‑publisher` or&nbsp;`dtslint`.

File: types/node/util.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ExE-Boss) [reply types/node/util.d.ts]: I’ve&nbsp;opened <https://github.com/microsoft/TypeScript/issues/36813> to&nbsp;add&nbsp;support for&nbsp;symbol&nbsp;properties to&nbsp;**TypeScript**&nbsp;namespaces.

**Comment by sandersn** [comment]
I restarted the failed CI jobs.

**Review by SimonSchick — APPROVED** [review]
Alright, this looks good, while the practical usability is still somewhat limited due to TS limitation the definitions are factually correct.

**Pull request merged**